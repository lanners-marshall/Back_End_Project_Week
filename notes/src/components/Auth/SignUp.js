import React from 'react';
import axios from 'axios';
import { Link } from 'react-router-dom';
import { Contain, MainH1, FlexForm, BTN, SubmitBtn, BTNDiv, Errors } from './css';

class SignUp extends React.Component {
	constructor(){
		super();
		this.state = {
			username: '',
			password: '',
			email: '',
			error: ''
		};
	}

	handleChange = event => {
 	  this.setState({[event.target.name]: event.target.value})
 	}

 	register = event => {
 		event.preventDefault()

 		const newUser = {
 			username: this.state.username,
 			password: this.state.password,
 			email: this.state.email
 		}

 		if (newUser.username.length < 5 || newUser.password.length < 5 || newUser.email.length < 5){
 			this.setState({
 				error: 'Please provide all information. Min length for each is 5 characters'
 			})
 		} else {

		 	axios.post('http://localhost:5555/users/register', newUser)
		 		.then(response => {
		 			console.log(response)
		 			localStorage.setItem('jwt', response.data.token);
		 			localStorage.setItem('loggedInAs', this.state.username);
		 			this.setState({
		 				error: ''
		 			})
		 			this.props.history.push('/notes')
		 		})
		 		.catch(error => {
		 			console.log(error)
		 			this.setState({
		 				error: error.response.data,
		 			})
		 		})

		 	const collaborator = {name: this.state.username}
		 	axios.post('http://localhost:5555/collaborators', collaborator)
		 		.then(response => {
		 			console.log(response)
		 		})
		 		.catch(error => {
		 			console.log(error)
		 		})
 		}
 	}

	render() {
		return (
			<div>
				<Contain>
					<MainH1>Sign Up Below</MainH1>
					<FlexForm>
						<input
							type="text"
							placeholder='username'
							onChange={this.handleChange}
							name="username"
							value={this.state.username}
						/>
						<input
							type="text"
							placeholder='email'
							onChange={this.handleChange}
							name="email"
							value={this.state.email}
						/>
						<input
							type="password"
							placeholder='password'
							onChange={this.handleChange}
							name="password"
							value={this.state.password}
						/>
						<SubmitBtn onClick={this.register}>Submit</SubmitBtn>
					</FlexForm>
				</Contain>
				<Link to='/signin'><BTNDiv><BTN>Need to go back click here!</BTN></BTNDiv></Link>
				<Errors>
					<p>{this.state.error}</p>
				</Errors>
			</div>
		)
	}
}

export default SignUp;

