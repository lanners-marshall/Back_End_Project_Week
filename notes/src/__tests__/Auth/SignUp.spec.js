import React from 'react';
import ReactDOM from 'react-dom';
import Enzyme, { shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { MemoryRouter, Route } from 'react-router-dom';

import SignUp from '../../components/Auth/SignUp';
import renderer from 'react-test-renderer';

describe('<SignUp />', () => {
  it('should match snapshot', () => {
	  const wrapper = shallow(
	  	<MemoryRouter initialEntries={['/signup']}>
	      <SignUp />
	    </MemoryRouter>
  	)
  	const tree = renderer.create(wrapper).toJSON()
  	expect(tree).toMatchSnapshot()
	})
});