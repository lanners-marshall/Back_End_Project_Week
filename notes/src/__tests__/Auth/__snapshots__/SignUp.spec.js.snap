// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignUp /> should match snapshot 1`] = `
<div>
  <div
    className="sc-bdVaJa hZjtPt"
  >
    <h1
      className="sc-bwzfXH grlnmy"
    >
      Sign Up Below
    </h1>
    <form
      className="sc-htpNat fvVwSE"
    >
      <input
        name="username"
        onChange={[Function]}
        placeholder="username"
        type="text"
        value=""
      />
      <input
        name="email"
        onChange={[Function]}
        placeholder="email"
        type="text"
        value=""
      />
      <input
        name="password"
        onChange={[Function]}
        placeholder="password"
        type="password"
        value=""
      />
      <button
        className="sc-ifAKCX iUOQTm"
        onClick={[Function]}
      >
        Submit
      </button>
    </form>
  </div>
  <a
    href="/signin"
    onClick={[Function]}
  >
    <div
      className="sc-EHOje hqEBXS"
    >
      <button
        className="sc-bxivhb eMpRCh"
      >
        Need to go back click here!
      </button>
    </div>
  </a>
  <div
    className="sc-bZQynM dZfyrm"
  >
    <p />
    <p />
  </div>
</div>
`;
